File: advent-of-code-2023/day1part1.exs
{:ok, content} = File.read("inputs/day1.txt")
lines = content |> String.split("\n", trim: true)

count = lines 
        |> Enum.map(fn line -> line
                               |> String.replace(~r/[^\d]/, "")
                               |> String.split("", trim: true) end)
        |> Enum.map(fn numbers -> "#{hd(numbers)}#{List.last(numbers)}" end)
        |> Enum.map(&Integer.parse(&1) |> elem(0))
        |> Enum.sum

IO.inspect count

File: advent-of-code-2024/src/bin/day1part1.rs
use std::fs;

fn main() {
    let file_path = "src/bin/inputs/day1part1";
    let contents = fs::read_to_string(file_path)
        .expect("Should have been able to read the file");

    let location_ids = contents
        .trim()
        .lines()
        .map(|line| line
            .split("   ")
            .map(|item| item.parse::<i32>().unwrap())
            .collect::<Vec<_>>())
        .collect::<Vec<_>>();

    let mut left_ids = vec![];
    let mut right_ids = vec![];
    for pair in location_ids.iter() {
        left_ids.push(pair[0]);
        right_ids.push(pair[1]);
    }

    left_ids.sort();
    right_ids.sort();

    let total = left_ids
        .into_iter()
        .zip(right_ids.into_iter())
        .map(|(l, r)| (l - r).abs())
        .sum::<i32>();

    println!("Total differences is {total}");
}